draft-ietf-ace-revoked-token-notification.md

```diff
@@ -215,6 +219,51 @@ At a high level, the steps of this protocol are as follows.
 
 {{sec-RS-examples}} provides examples of the protocol flow and message exchanges between the AS and a registered device.
 
+# Issuing of Access Tokens at the AS # {#sec-issuing-access-tokens-as}
+
+An AS that supports the method defined in this document MUST adhere to the following rules when issuing an access token.
+
+* All the intended header parameters in the access token MUST be specified within integrity-protected fields.
+
+* If the access token is a CWT, the following applies.
+
+  - Any "unprotected" field MUST be empty, i.e., its value MUST be encoded as the empty CBOR map (0xa0). This applies to: the top-level "unprotected" field of the COSE object used for the CWT; the "unprotected" field of each element of the "signatures" array; and the "unprotected" field of each element of any "recipients" array (see {{Sections 2, 3, 4, 5, and 6 of RFC9052}}).
+
+  - Consistent with the specific COSE object used for the CWT, the corresponding tagged structure in the set COSE_Tagged_Message MUST be used (see {{Section 2 of RFC9052}}). That is, the CBOR array that encodes the CWT MUST be tagged by using the COSE CBOR tag corresponding to the used COSE object. Table 1 in {{Section 2 of RFC9052}} specifies the tag numbers in question.
+
+    In turn, the resulting tagged data item MUST be tagged by using the CWT CBOR tag with tag number 61 (see {{Section 6 of RFC8392}}). After that, the resulting data item MUST NOT be further tagged.
+
+    Encoding of the tag numbers MUST be done using definite lengths, and the length of the encoded tag numbers MUST be the minimum possible length. This means that the tag number 16 is encoded as 0xd0 and not as 0xd810.
+
+    The example in {{fig-cwt-cose-encrypt0}} shows a CWT that uses the COSE object COSE_Encrypt0 (see {{Section 5.2 of RFC9052}}).
+
+* If the access token is a JWT, the following applies.
+
+  - If the access token is specifically a JWS {{RFC7515}} that uses the JWS JSON Serialization, it MUST NOT include the JWS Unprotected Header.
```

Comment:
A JWT is never allowed to be in JSON serialization.

> JWTs are always represented using the JWS Compact Serialization or the JWE Compact Serialization.

- https://datatracker.ietf.org/doc/html/rfc7519#section-1

Reply:
Thanks for this comment! It is now addressed at: https://github.com/ace-wg/ace-revoked-token-notification/pull/13/commits/ab91b71c652ff37ee96c8632071103ee4b8f4232

The JWS Compact Serialization from RFC 7515 and the JWE Compact Serialization from RFC 7516 do not admit unprotected headers altogether (see Section 3.1 of both RFCs).

This means that JWTs as defined in RFC 7519 are basically just fine. The explicit requirements on avoiding unprotected headers are needed only for possible other access tokens that use the JWS/JWE JSON Serialization.

Now the text in Sections 3 and 11.1 reflects that.